File: src/test/testData/java/some/pkg/MyClass.java
public class MyClass {
    // #1
    public String greeting() {
        return "Hello, world!";
    }

    // #2
    public String farewell(MyInnerClass fakeChild, List<String> names) {
        return "Goodbye, cruel world!";
    }
    
    class MyInnerClass {
        // #3
        public String innerGreeting() {
            return "Hello, inner!";
        }

        class MyInnerInnerClass {
            // #4
            public String innerInnerGreeting() {
                return "Hello, inner inner!"
            }

            // #5
            public String innerInnerFarewell() {
                return "Goodbye, cruel inner inner!";
            }
        }

        // #6
        public String innerFarewell() {
            return "Goodbye, cruel inner!";
        }
    }
    
    static class MyStaticInnerClass {
        // #7
        public String staticInnerGreeting() {
            return "Hello, static inner!";
        }

        // #8
        public String staticInnerFarewell() {
            return "Goodbye, cruel static inner!";
        }
    }

    // #9
    public static String generateStatic() {
        return "&)*(<$&Y*()$&*()<"
    }
}        

private class PrivateTopLevelClass {
    // #10
    public String privateTopLevelGreeting() {
        return "Hello, private top level!";
    }

    // #11
    public String privateTopLevelFarewell() {
        return "Goodbye, cruel private top level!";
    }
}

File: src/test/testData/kotlin/some/pkg/MyClass.kt
class MyClass {
    // #1
    fun greeting(): String {
        return "Hello, world!"
    }

    // #2
    fun farewell(fakeChild: myInnerClass?, names: List<String>): String {
        return "Goodbye, cruel world!"
    }

    internal inner class MyInnerClass {
        // #3
        fun innerGreeting(): String {
            return "Hello, inner!"
        }

        internal inner class MyInnerInnerClass {
            // #4
            fun innerInnerGreeting(): String {
                return "Hello, inner inner!"
            }

            // #5
            fun innerInnerFarewell(): String {
                return "Goodbye, cruel inner inner!"
            }
        }

        // #6
        fun innerFarewell(): String {
            return "Goodbye, cruel inner!"
        }
    }

    internal class MyStaticInnerClass {
        // #7
        fun innerGreeting(): String {
            return "Hello, static inner!"
        }

        // #8
        fun innerFarewell(): String {
            return "Goodbye, cruel static inner!"
        }
    }

    companion object {
        // #9
        fun generateStatic(): String {
            return "&)*(<$&Y*()$&*()<"
        }
    }
}

private class PrivateTopLevelClass {
    // #10
    fun privateTopLevelGreeting(): String {
        return "Hello, private top level!"
    }

    // #11
    fun privateTopLevelFarewell(): String {
        return "Goodbye, cruel private top level!"
    }
}

// # 12
fun topLevelFunction(): String = "Hello. top level!"

